src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp

```diff
@@ -840,7 +839,11 @@ ConvSolution ConvHipImplicitGemmBwdDataV4R1Xdlops::GetSolution(
 {
     ConvSolution result;
 
-    assert(config.IsValid(ctx));
+    if(!config.IsReallyValid(ctx))
```

Comment:
[Recommendation] `GetSolution()` is not allowed to fail if `IsApplicable()` is true, unless there is a bug in the Solver (see doc in solver.hpp). If you see possibility of this, and would like to double-check, then:
- hide all checks under `#if !NDEBUG` so this will happen only in Debug builds and wouldn't waste time in releases.
- Use MIOPEN_THROW to terminate as this is internal error.

Of course you can keep this as is, if you think that probability of such errors is high. But I would say that the Solvers that have high probability of failure in GetSolution are not of production quality yet and should be disabled by default.

Reply:
Current IsApplicable() will call EuristicInit(), which will loop over all tuning range to try to find one set of parameters. If IsApplicable() will return true if EuristicInit() find THAT set of parameters. So GetSolution() should never fail, if IsApplicable() is true. So, this assert in GetSolution() is not supposed to be hit.

Reply:
Yes, thanks, of course I know that. I give recommendation, you decide.

Reply:
@asroy If you decide to follow the resomendation, please use `#if !MIOPEN_NDEBUG`, _NOT_ `#if !NDEBUG` (this is incorrect).